# **TikTok Project**
**Course 4 - The Power of Statistics**

You are a data professional at TikTok. The current project is reaching its midpoint; a project proposal, Python coding work, and exploratory data analysis have all been completed.

The team has reviewed the results of the exploratory data analysis and the previous executive summary the team prepared. You received an email from Orion Rainier, Data Scientist at TikTok, with your next assignment: determine and conduct the necessary hypothesis tests and statistical analysis for the TikTok classification project.

A notebook was structured and prepared to help you in this project. Please complete the following questions.


# **Course 4 End-of-course project: Data exploration and hypothesis testing**

In this activity, you will explore the data provided and conduct hypothesis testing.
<br/>

**The purpose** of this project is to demostrate knowledge of how to prepare, create, and analyze hypothesis tests.

**The goal** is to apply descriptive and inferential statistics, probability distributions, and hypothesis testing in Python.
<br/>

*This activity has three parts:*

**Part 1:** Imports and data loading
* What data packages will be necessary for hypothesis testing?

**Part 2:** Conduct hypothesis testing
* How will descriptive statistics help you analyze your data?

* How will you formulate your null hypothesis and alternative hypothesis?

**Part 3:** Communicate insights with stakeholders

* What key business insight(s) emerge from your hypothesis test?

* What business recommendations do you propose based on your results?

<br/>

Follow the instructions and answer the questions below to complete the activity. Then, complete an executive summary using the questions listed on the PACE Strategy Document.

Be sure to complete this activity before moving on. The next course item will provide you with a completed exemplar to compare to your own work.



# **Data exploration and hypothesis testing**

<img src="images/Pace.png" width="100" height="100" align=left>

# **PACE stages**

Throughout these project notebooks, you'll see references to the problem-solving framework PACE. The following notebook components are labeled with the respective PACE stage: Plan, Analyze, Construct, and Execute.

<img src="images/Plan.png" width="100" height="100" align=left>


## **PACE: Plan**

Consider the questions in your PACE Strategy Document and those below to craft your response.

1. What is your research question for this data project? Later on, you will need to formulate the null and alternative hypotheses as the first step of your hypothesis test. Consider your research question now, at the start of this task.

Is there a relationshiop between view count and the verified / unverified status of accounts.

*Complete the following steps to perform statistical analysis of your data:*

### **Task 1. Imports and Data Loading**

Import packages and libraries needed to compute descriptive statistics and conduct a hypothesis test.

<details>
  <summary><h4><strong>Hint:</strong></h4></summary>

Be sure to import `pandas`, `numpy`, `matplotlib.pyplot`, `seaborn`, and `scipy`.

</details>

In [1]:
# Import packages for data manipulation
import pandas as pd
from scipy import stats


Load the dataset.

**Note:** As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

In [2]:
# Load dataset into dataframe
data = pd.read_csv("tiktok_dataset.csv")

<img src="images/Analyze.png" width="100" height="100" align=left>

<img src="images/Construct.png" width="100" height="100" align=left>

## **PACE: Analyze and Construct**

Consider the questions in your PACE Strategy Document and those below to craft your response:
1. Data professionals use descriptive statistics for Exploratory Data Analysis. How can computing descriptive statistics help you learn more about your data in this stage of your analysis?


Descriptive stats help us get a brief overview of the data. We can then use this to further drill down and get to know more about the data.

### **Task 2. Data exploration**

Use descriptive statistics to conduct Exploratory Data Analysis (EDA).



<details>
  <summary><h4><strong>Hint:</strong></h4></summary>

Refer back to *Self Review Descriptive Statistics* for this step-by-step proccess.

</details>

Inspect the first five rows of the dataframe.

In [3]:
data.head()


# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

In [4]:
data.describe(include="all")

# claim_status      video_id  video_duration_sec  \
count   19382.000000        19084  1.938200e+04        19382.000000   
unique           NaN            2           NaN                 NaN   
top              NaN        claim           NaN                 NaN   
freq             NaN         9608           NaN                 NaN   
mean     9691.500000          NaN  5.627454e+09           32.421732   
std      5595.245794          NaN  2.536440e+09           16.229967   
min         1.000000          NaN  1.234959e+09            5.000000   
25%      4846.250000          NaN  3.430417e+09           18.000000   
50%      9691.500000          NaN  5.618664e+09           32.000000   
75%     14536.750000          NaN  7.843960e+09           47.000000   
max     19382.000000          NaN  9.999873e+09           60.000000   

                                 video_transcription_text verified_status  \
count                                               19084           19382   
unique                                              19012               2   
top     a friend read  in the media a claim that badmi...    not verified   
freq                                                    2           18142   
mean                                                  NaN             NaN   
std                                                   NaN             NaN   
min                                                   NaN             NaN   
25%                                                   NaN             NaN   
50%                                                   NaN             NaN   
75%                                                   NaN             NaN   
max                                                   NaN             NaN   

       author_ban_status  video_view_count  video_like_count  \
count              19382      19084.000000      19084.000000   
unique                 3               NaN               NaN   
top               active               NaN               NaN   
freq               15663               NaN               NaN   
mean                 NaN     254708.558688      84304.636030   
std                  NaN     322893.280814     133420.546814   
min                  NaN         20.000000          0.000000   
25%                  NaN       4942.500000        810.750000   
50%                  NaN       9954.500000       3403.500000   
75%                  NaN     504327.000000     125020.000000   
max                  NaN     999817.000000     657830.000000   

        video_share_count  video_download_count  video_comment_count  
count        19084.000000          19084.000000         19084.000000  
unique                NaN                   NaN                  NaN  
top                   NaN                   NaN                  NaN  
freq                  NaN                   NaN                  NaN  
mean         16735.248323           1049.429627           349.312146  
std          32036.174350           2004.299894           799.638865  
min              0.000000              0.000000             0.000000  
25%            115.000000              7.000000             1.000000  
50%            717.000000             46.000000             9.000000  
75%          18222.000000           1156.250000           292.000000  
max         256130.000000          14994.000000          9599.000000

Check for and handle missing values.

In [13]:
data.isna().sum()

#                             0
claim_status                298
video_id                      0
video_duration_sec            0
video_transcription_text    298
verified_status               0
author_ban_status             0
video_view_count            298
video_like_count            298
video_share_count           298
video_download_count        298
video_comment_count         298
dtype: int64

In [16]:
non_null = data.dropna(axis = 0)


In [17]:
non_null

# claim_status    video_id  video_duration_sec  \
0          1        claim  7017666017                  59   
1          2        claim  4014381136                  32   
2          3        claim  9859838091                  31   
3          4        claim  1866847991                  25   
4          5        claim  7105231098                  19   
...      ...          ...         ...                 ...   
19079  19080      opinion  1492320297                  49   
19080  19081      opinion  9841347807                  23   
19081  19082      opinion  8024379946                  50   
19082  19083      opinion  7425795014                   8   
19083  19084      opinion  4094655375                  58   

                                video_transcription_text verified_status  \
0      someone shared with me that drone deliveries a...    not verified   
1      someone shared with me that there are more mic...    not verified   
2      someone shared with me that american industria...    not verified   
3      someone shared with me that the metro of st. p...    not verified   
4      someone shared with me that the number of busi...    not verified   
...                                                  ...             ...   
19079  in our opinion the earth holds about 11 quinti...    not verified   
19080  in our opinion the queens in ant colonies live...    not verified   
19081  in our opinion the moon is moving away from th...    not verified   
19082  in our opinion lightning strikes somewhere on ...    not verified   
19083  in our opinion a pineapple plant can only prod...    not verified   

      author_ban_status  video_view_count  video_like_count  \
0          under review          343296.0           19425.0   
1                active          140877.0           77355.0   
2                active          902185.0           97690.0   
3                active          437506.0          239954.0   
4                active           56167.0           34987.0   
...                 ...               ...               ...   
19079            active            6067.0             423.0   
19080            active            2973.0             820.0   
19081            active             734.0             102.0   
19082            active            3394.0             655.0   
19083            active            5034.0             815.0   

       video_share_count  video_download_count  video_comment_count  
0                  241.0                   1.0                  0.0  
1                19034.0                1161.0                684.0  
2                 2858.0                 833.0                329.0  
3                34812.0                1234.0                584.0  
4                 4110.0                 547.0                152.0  
...                  ...                   ...                  ...  
19079               81.0                   8.0                  2.0  
19080               70.0                   3.0                  0.0  
19081                7.0                   2.0                  1.0  
19082              123.0                  11.0                  4.0  
19083              281.0                  11.0                  1.0  

[19084 rows x 12 columns]

You are interested in the relationship between `verified_status` and `video_view_count`. One approach is to examine the mean value of `video_view_count` for each group of `verified_status` in the sample data.

In [19]:
data.groupby("verified_status").video_view_count.mean()

verified_status
not verified    265663.785339
verified         91439.164167
Name: video_view_count, dtype: float64

### **Task 3. Hypothesis testing**

Before you conduct your hypothesis test, consider the following questions where applicable to complete your code response:

1. Recall the difference between the null hypothesis and the alternative hypotheses. What are your hypotheses for this data project?



Your goal in this step is to conduct a two-sample t-test. Recall the steps for conducting a hypothesis test:


1.   State the null hypothesis and the alternative hypothesis
2.   Choose a signficance level
3.   Find the p-value
4.   Reject or fail to reject the null hypothesis



Null hypothesis - The mean value for view count is the same for verified and unverified
Alternate hypothesis - The mean value for view count for verified is not the same as unverified


In [28]:
verified = non_null[non_null["verified_status"] == "verified"]
unverified = non_null[non_null["verified_status"] == "not verified"]


In [29]:
verified.shape

(1200, 12)

In [30]:
unverified.shape

(17884, 12)

You choose 5% as the significance level and proceed with a two-sample t-test.

In [32]:
verified

# claim_status    video_id  video_duration_sec  \
9         10        claim  4660861094                  45   
76        77        claim  2342638585                  31   
105      106        claim  7687467484                  49   
117      118        claim  2115386652                  48   
140      141        claim  6296788623                  16   
...      ...          ...         ...                 ...   
19021  19022      opinion  4696592896                  41   
19022  19023      opinion  8542621111                  17   
19061  19062      opinion  6326142814                  28   
19063  19064      opinion  9390713852                  36   
19074  19075      opinion  4899891756                  48   

                                video_transcription_text verified_status  \
9      someone shared with me that it would take a 50...        verified   
76     a friend shared with a friend that only one-fo...        verified   
105    a friend mentioned someone discovered that lon...        verified   
117    a friend mentioned someone discovered that bee...        verified   
140    a friend reported someone discovered that shan...        verified   
...                                                  ...             ...   
19021  in our opinion koalas sleep up to 22 hours per...        verified   
19022    in our opinion some sharks can glow in the dark        verified   
19061  in our opinion most children in the world are ...        verified   
19063  in our opinion humans are more likely to have ...        verified   
19074  in our opinion the average american spends tim...        verified   

      author_ban_status  video_view_count  video_like_count  \
9                active          931587.0          171051.0   
76               active          677288.0          214708.0   
105        under review          631075.0          284311.0   
117              active          915553.0          272536.0   
140              active          775229.0           42657.0   
...                 ...               ...               ...   
19021            active            2118.0             465.0   
19022            active            5195.0            1019.0   
19061            active            1607.0             321.0   
19063            active            9084.0            2473.0   
19074            active            5958.0             840.0   

       video_share_count  video_download_count  video_comment_count  
9                67739.0                4104.0               2540.0  
76               66446.0                2167.0               1428.0  
105              97442.0                1163.0                354.0  
117              78299.0                   7.0                  3.0  
140               1636.0                 835.0                247.0  
...                  ...                   ...                  ...  
19021               57.0                   5.0                  2.0  
19022              245.0                  20.0                  6.0  
19061                8.0                   6.0                  1.0  
19063              384.0                  58.0                 14.0  
19074              219.0                  19.0                  1.0  

[1200 rows x 12 columns]

In [33]:
# Conduct a two-sample t-test to compare means
### YOUR CODE HERE ###
stats.ttest_ind(verified["video_view_count"],unverified["video_view_count"],equal_var = True)

Ttest_indResult(statistic=-18.250939509545827, pvalue=8.632160883925904e-74)

**Question:** Based on the p-value you got above, do you reject or fail to reject the null hypothesis?


Our p value is = 8.63 * 10^(-74)
This is significantly smaller than our significant value of 0.05
So we can say that the p value is smaller so we reject the null hypothesis

<img src="images/Execute.png" width="100" height="100" align=left>

## **PACE: Execute**

Consider the questions in your PACE Strategy Documentto reflect on the Execute stage.

## **Step 4: Communicate insights with stakeholders**

*Ask yourself the following questions:*

1. What business insight(s) can you draw from the result of your hypothesis test?

So now that we reject the null hypothesis
We can conclude that the means of verified and unverified groups are not the same
Thus there may be some relationship between the two variables which has may be analysed more closely.

Based on the preliminary data, it seems that unverified author videos have far more views
So the relationship may be unverified authors implies a greater number of views on the authors view

**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.